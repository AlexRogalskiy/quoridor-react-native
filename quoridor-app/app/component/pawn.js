import React from "react";
import { Text, StyleSheet } from "react-native";

const styles = StyleSheet.create({
  pawn: {
    position: "absolute",
    fontSize: 30
  }
});

export class Pawn extends React.Component {
  render() {
    return (
      <Text
        style={[
          styles.pawn,
          {
            color: this.props.color,
            top: this.props.position.top,
            left: this.props.position.left
          }
        ]}
      >
        &#9823;
      </Text>
    );
  }
}
