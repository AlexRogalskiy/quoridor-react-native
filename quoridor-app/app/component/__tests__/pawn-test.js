import React from "react";
import renderer from "react-test-renderer";
import { Text, StyleSheet } from "react-native";

import { Pawn } from "../pawn";

it("renders correctly", () => {
  const tree = renderer
    .create(<Pawn color="red" position="{top: o, left: 10}" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("<Pawn />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<Pawn color="red" />).toJSON();
    expect(tree.children.length).toBe(1);
  });
});
