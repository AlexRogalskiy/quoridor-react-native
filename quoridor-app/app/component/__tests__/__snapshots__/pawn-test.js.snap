// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 30,
        "position": "absolute",
      },
      Object {
        "color": "red",
        "left": undefined,
        "top": undefined,
      },
    ]
  }
>
  â™Ÿ
</Text>
`;
